actors/builtin/market/market_state.go

```diff
@@ -205,200 +205,204 @@ func (st *State) generateStorageDealID() abi.DealID {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-// Balance table operations
+// Method utility functions
 ////////////////////////////////////////////////////////////////////////////////
 
-func (st *State) MutateBalanceTable(s adt.Store, c *cid.Cid, f func(t *adt.BalanceTable) error) error {
-	t, err := adt.AsBalanceTable(s, *c)
-	if err != nil {
-		return err
-	}
-
-	if err := f(t); err != nil {
-		return err
-	}
-
-	rc, err := t.Root()
-	if err != nil {
-		return err
+func (st *State) mustGetDeal(rt Runtime, proposals *DealArray, dealID abi.DealID) *DealProposal {
```

Comment:
FYI: for the future I'd prefer this function to disappear, or move into the actor code as a free function. State should not use `rt`.